### PR TITLE
Removed unused properties from BackstagePaletteAdditions

### PR DESCRIPTION
**Pull request by awanlin**
Signed-off-by: Andre Wanlin <awanlin@rapidrtc.com>

## Hey, I just made a Pull Request!

Removed the following unused properties from BackstagePaletteAdditions under the `bursts` property:

```ts
slackChannelText: string;
backgroundColor: {
    default: string;
};
```

From what I could tell these are not being used and by removing them will keep this object cleaner

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/serious-houses-watch.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+'@backstage/theme': patch
```
Comment (jhaals) [inline .changeset/serious-houses-watch.md:2]: Should be a minor bump even if there are no usages in the repository.

File: .changeset/serious-houses-watch.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+---
+
+Removed the following unused properties from BackstagePaletteAdditions under the `bursts` property:
```
Comment (jhaals) [inline .changeset/serious-houses-watch.md:5]:
```suggestion
**BREAKING**: Removed the following unused properties from BackstagePaletteAdditions under the `bursts` property:
```

**Review by jhaals — COMMENTED** [review]
Thanks for cleaning things up @awanlin! 🏅 

Some small tweaks needed. Looks good to me otherwise.

**Comment by benjdlambert** [comment]
Maybe we could actually remove the entire burst block here actually @awanlin, I think that it's legacy and just hanging around. Could you maybe investigate that and see if there's anything that depends on it?

File: .changeset/serious-houses-watch.md (hunk 2 above)
Comment (awanlin) [reply .changeset/serious-houses-watch.md]: This has been added, thanks 👍

File: .changeset/serious-houses-watch.md (hunk 1 above)
Comment (awanlin) [reply .changeset/serious-houses-watch.md]: I've corrected this to be minor

**Comment by Rugvip** [comment]
@awanlin Would be great to get rid of the entire `bursts` section tbh, also feeling it's best to do all at once to minimize impact a bit. Seeing as theme is a pretty central piece that's propagated through a React context I'd say we should start with a deprecation. We'd mark the entire `bursts` section as `@deprecated` in the type, and I'd say we could go straight for a warning in the implementations too. Something like this:

```ts
get bursts() {
  if (!warned) {
    console.warn('DEPRECATION WARNING: ...');
    warned = true;
    return actualBursts;
  }
}
```

To replace the existing usages, I think `bursts.gradient.linear` should be possible to replace using styles from `theme.page`, or maybe an explicit `theme.getPageTheme({ themeId: 'home' })`. The font color might be trickier, but I think it might also be alright to just replace that with the primary text color?

**Comment by awanlin** [comment]
@Rugvip I'm working on what you suggested, have most of it in place, the font color replacement has been a bit trickery than expected. Also, I'm not exactly sure where to add the code snippet you suggested?

**Comment by Rugvip** [comment]
@awanlin Nice! 👍 

The getter might be a bit tricky to get in there in a what where it doesn't get copied to a plain prop. Might work to add it at https://github.com/backstage/backstage/blob/928726fc1fa057f153d372d1f217f7486dde2882/packages/theme/src/themes.ts#L40, or it might need to be injected at https://github.com/backstage/backstage/blob/928726fc1fa057f153d372d1f217f7486dde2882/packages/theme/src/baseTheme.ts#L293. Worst case it might be that a `Object.defineProperty` is needed.

The purpose of the thing would be to log the warning if anyone tries to access to property. The trap is that the property might be read and copied into a separate object, and that the accessor is lost when that happens. Try this piece in the console for example:

```js
o = {
  get x() {
    console.log('Read x');
    return 'x';
  }
};
console.log('1', o.x);
console.log('2', o.x);
console.log('copying');
o2 = { ...o };
console.log('copied');
console.log('4', o2.x);
console.log('5', o2.x);
```

**Comment by Rugvip** [comment]
@awanlin nice, thank you! 😁 🎉 

Yeah agree we've got a gap that needs to be filled. I think reasonable options would be to add a page theme header color to the base palette and/or each page theme.

Thinking for example this addition to the root of the palette:

```ts
textPageThemeContrast?: string
```

Or whatever name makes sense 😅 

And then we could perhaps pair that up with an addition to the `PageTheme` type:

```ts
fontColor?: string
```

and have that default to white in `genPageTheme`.

Tbh just to minimize additions for now I think it might be best to stick to only the `PageTheme` update, don't necessarily need to root color if we default it anyway.

On a separate note, the current signature for `getPageTheme` is `(colors?: string[], shape: string)` I'm thinking it would be best to convert that to an options object -> `(options: { colors: string[], shape: string, fontColor?: string })`. With backwards compatibility if you feel like it, but a bit optional since the theme package isn't stable yet

**Comment by awanlin** [comment]
I'm still working on this, things just got busy on my end

File: .changeset/serious-houses-watch.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/theme': patch
+---
+
+**DEPRECIATED**: The `bursts` object from BackstagePaletteAdditions has been depreciated and will be removed in a future release
```
[the remark this answers is not in the document]
Comment (awanlin) [reply .changeset/serious-houses-watch.md]: Thanks, fixed it

**Comment by awanlin** [comment]
@Rugvip, finally, got back onto this. I've made the changes that you suggested by adding just a `fontColor` to `PageTheme` and then updated `genPageTheme()` accordingly with the new signature and defaulting `fontColor` to white.

Could use some help with the best way to describe this? The changes are across a few packages so I'll need some more changesets I think.

Also, this change impacts #11769. Should I update the changeset to note it doesn't use `theme.palette.bursts.fontColor` but now uses `theme.page.fontColor`? CC @benjdlambert

File: .changeset/serious-houses-watch.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/theme': patch
+---
+
+**DEPRECATED**: The `bursts` object from `BackstagePaletteAdditions` has been depreciated and will be removed in a future release
```
Comment (Rugvip) [inline .changeset/serious-houses-watch.md:5]: I think in addition we can mention that `genPageTheme` now has a third argument that allows you to set the font color, but it defaults to white.

File: packages/theme/src/pageTheme.ts [hunk 5]
```diff
@@ -63,12 +64,25 @@ export const colorVariants: Record<string, string[]> = {
  * As the background shapes and colors are decorative, we place them onto the
  * page as a css background-image instead of an html element of its own.
  */
-export function genPageTheme(colors: string[], shape: string): PageTheme {
-  const gradientColors = colors.length === 1 ? [colors[0], colors[0]] : colors;
+export function genPageTheme(options: {
+  colors: string[];
+  shape: string;
+  fontColor?: string;
```
Comment (Rugvip) [inline packages/theme/src/pageTheme.ts:70]: Let's make the 3rd argument an optional options object, just in case we want to add more stuff in the future

**Review by Rugvip — COMMENTED** [review]
Nice, thank you! 👍 

This just a small addition to the existing changeset is enough, along with something short along the lines of "Updated to remove usage of the `bursts` colors in the theme palette" for `core-components` and `catalog`. The `org` plugin doesn't need a changeset since stories don't affect the published package.

**Comment by Rugvip** [comment]
Regarding the existing changeset, yep let's update it

File: packages/theme/src/pageTheme.ts (hunk 5 above)
Comment (awanlin) [reply packages/theme/src/pageTheme.ts]: Sounds good, done 👍

File: .changeset/serious-houses-watch.md (hunk 4 above)
Comment (awanlin) [reply .changeset/serious-houses-watch.md:5]: I've added a note about this to the changeset 👍

**Comment by awanlin** [comment]
Got the latest round of changes in place, ready for another review @Rugvip when you have some time, please

**Comment by awanlin** [comment]
Sorry to pester you @Rugvip @freben @jhaals but would we be able to merge this? It's getting a little long and would be nice to square this away

**Review by freben — APPROVED** [review]
Let's go!

**Comment by awanlin** [comment]
Thanks for the merge @freben and many thanks to @jhaals, @benjdlambert, and @Rugvip !!!

**Pull request closed without merging**